packages/terra-date-picker/src/DatePicker.jsx

```diff
@@ -449,11 +449,7 @@ class DatePicker extends React.Component {
         }
       }
     } else {
-      selectedDateInPicker = this.state.selectedDate;
-    }
-
-    if (this.isDefaultDateAcceptable && !this.isDateWithinRange(selectedDateInPicker)) {
-      selectedDateInPicker = undefined;
+      selectedDateInPicker = (this.isDefaultDateAcceptable && !this.isDateWithinRange(selectedDateInPicker)) ? undefined : this.state.selectedDate;
```

Comment:
This is not the way we'll want to handle the intent of the originating request. 

The current date-picker behavior is functioning as designed, keeping only valid dates as selected, and the calendar correspondingly showing a valid selected date, otherwise we assume for the current date. I will address alternative approaches on the originating Jira, and we will need to decide on how best to solve with a new tech design.

Reply:
Applied the alternative approach suggestion to the originating Jira for review:
https://jira2.cerner.com/browse/UXPLATFORM-8001?focusedCommentId=11859086&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-11859086

This PR can be closed as do not merge and won't fix.